src/components/Articles/ViewArticle.js

```diff
@@ -1,41 +1,157 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import { fetchSpecificArticle } from '../../actions/articleActions';
+import PropTypes from 'prop-types';
+import {
+  fetchSpecificArticle,
+  updateArticle,
+} from '../../actions/articleActions';
 import ReadArticle from './ReadArticle';
+import NewArticleForm from './NewArticleForm';
 
 export class ViewArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      body: '',
+      isEditing: false,
+    };
+    this.toggleEdit = this.toggleEdit.bind(this);
+  }
+
   componentDidMount() {
-    const { match: { params: { slug } }, fetchSpecificArticle } = this.props;
+    const { fetchSpecificArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
     fetchSpecificArticle(slug);
   }
 
-  render() {
-    const { articlePayload, articlePayload: { article } } = this.props;
+  componentWillReceiveProps(nextProps) {
+    const { articlePayload } = nextProps;
+    const { article: { title, body, description } } = articlePayload;
+    this.setState({ title });
+    this.setState({ body });
+    this.setState({ description });
+
+    if (nextProps.editArticleSuccess === true) {
+      this.setState({ isEditing: false });
+      const { history } = this.props;
+      history.push('/dashboard');
+    }
+  }
+
+  handleChange = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleEditorChange = value => {
+    this.setState({ body: value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const { title, description, body } = this.state;
+    const payload = {
+      article: {
+        title,
+        description,
+        body,
+      },
+    };
 
+    const { updateArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    updateArticle(slug, payload);
+  };
+
+  resetForm = () => {
+    this.setState({
+      title: '',
+      description: '',
+      body: '',
+    });
+  };
+
+  toggleEdit() {
```

Comment:
Consider making this an arrow function which binds automatically to the class. Then remove line `21`.

Reply:
Thanks, @kakaemma `toggleEdit()` has been redefined as an arrow function and line `21` deleted.